src/core/classes/erc-721-dropable.ts

```diff
@@ -0,0 +1,247 @@
+import { TokensClaimedEvent } from "contracts/Drop";
+import { BigNumber, BigNumberish, BytesLike, ethers, utils } from "ethers";
+import { prepareClaim } from "../../common/claim-conditions";
+import { FEATURE_NFT_DROPABLE } from "../../constants/erc721-features";
+import { SmartContract } from "../../contracts/smart-contract";
+import {
+  CommonNFTInput,
+  NFTMetadata,
+  NFTMetadataInput,
+  NFTMetadataOwner,
+} from "../../schema";
+import { TokensLazyMintedEvent } from "contracts/LazyMintERC721";
+import { ClaimVerification, UploadProgressEvent } from "../../types";
+import { BaseDropERC721 } from "../../types/eips";
+import { DetectableFeature } from "../interfaces/DetectableFeature";
+import { IStorage } from "../interfaces/IStorage";
+import { TransactionResultWithId } from "../types";
+import { ContractMetadata } from "./contract-metadata";
+import { ContractWrapper } from "./contract-wrapper";
+import { DropClaimConditions } from "./drop-claim-conditions";
+import { Erc721 } from "./erc-721";
+
+/**
+ * Lazily mint and claim ERC721 NFTs
+ * @remarks Manage claim phases and claim ERC721 NFTs that have been lazily minted.
+ * @example
+ * ```javascript
+ * const contract = await sdk.getContract("{{contract_address}}");
+ * await contract.drop.claim(quantity);
+ * ```
+ */
+export class Erc721Dropable implements DetectableFeature {
+  featureName = FEATURE_NFT_DROPABLE.name;
+
+  /**
+   * Configure claim conditions
+   * @remarks Define who can claim NFTs in the collection, when and how many.
+   * @example
+   * ```javascript
+   * const presaleStartTime = new Date();
+   * const publicSaleStartTime = new Date(Date.now() + 60 * 60 * 24 * 1000);
+   * const claimConditions = [
+   *   {
+   *     startTime: presaleStartTime, // start the presale now
+   *     maxQuantity: 2, // limit how many mints for this presale
+   *     price: 0.01, // presale price
+   *     snapshot: ['0x...', '0x...'], // limit minting to only certain addresses
+   *   },
+   *   {
+   *     startTime: publicSaleStartTime, // 24h after presale, start public sale
+   *     price: 0.08, // public sale price
+   *   }
+   * ]);
+   * await contract.claimConditions.set(claimConditions);
+   * ```
+   */
+  public claimConditions: DropClaimConditions<BaseDropERC721>;
```

Comment:
so this is not optional right now, but it should be after the contract changes right? can you add a //TODO in there?

Reply:
This shouldn't be optional right - `IDrop` still depends on `IClaimCondition` which I think makes sense + `setClaimCondition` lives in `IDrop` so we couldn't really separate it out, are we making additionally interface changes besides whats on main? 

https://github.com/thirdweb-dev/contracts/blob/8e831c04dc38d761f7533ffe743c4d40e8f9a393/contracts/feature/interface/IDrop.sol#L51

Reply:
I guess as per your notion doc, it does make sense to split them out in the future